middleware/forwarded.go

```diff
@@ -0,0 +1,42 @@
+package middleware
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+)
+
+// extractHost returns the Host IP address without any port information
+func extractHost(address string) string {
+	hostIP := net.ParseIP(address)
+	if hostIP != nil {
+		return hostIP.String()
+	}
+	var err error
+	hostStr, _, err := net.SplitHostPort(address)
+	if err != nil {
+		// Invalid IP address, just return it so it shows up in the logs
```

Comment:
What would be an example of an invalid _IP address_ that gets us here?

Reply:
This was added first to catch the proxy making a mistake and returned something invalid like a hostname. The goal of this PR is to add something to diagnose incorrect sending of samples in Cortex. Additionally the `Forwarded` header supports Obfuscated Identifiers (see https://tools.ietf.org/html/rfc7239#section-6.3) which are strings.